instrumentation/net/http/otelhttp/internal/semconv/util.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package semconv // import "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/semconv"
+
+import (
+	"net"
+	"strconv"
+	"strings"
+)
+
+// splitHostPort splits a network address hostport of the form "host",
+// "host%zone", "[host]", "[host%zone], "host:port", "host%zone:port",
+// "[host]:port", "[host%zone]:port", or ":port" into host or host%zone and
+// port.
+//
+// An empty host is returned if it is not provided or unparsable. A negative
+// port is returned if it is not provided or unparsable.
+func splitHostPort(hostport string) (host string, port int) {
```

Comment:
Is this only used for unit tests?  If so, it could be in a _test.go file.

Reply:
This is used by the [duplicate semconv](https://github.com/MadVikingGod/opentelemetry-go-contrib/blob/30adb7e838d73484398ef40dfc910b6ef690f5fa/instrumentation/net/http/otelhttp/internal/semconv/dup.go#L57) and [1.24.0](https://github.com/MadVikingGod/opentelemetry-go-contrib/blob/30adb7e838d73484398ef40dfc910b6ef690f5fa/instrumentation/net/http/otelhttp/internal/semconv/v1.24.0.go#L52).  I did remove the test accidentally, I will add that back.